### PR TITLE
Enable SpatialGDK Tests on MacOS

### PR DESCRIPTION
**Pull request by jessicafalk**
#### Description
enabling SpatialGDK tests for macOS CI and also moved the test repo to be the GDKTestGyms instead for macOS. 

#### Tests
corresponding build: https://buildkite.com/improbable/unrealgdk-premerge/builds/14995#_

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/SpatialOSDispatcherInterface/SpatialOSDispatcherSpy.cpp [hunk 1]
```diff
@@ -1,7 +1,5 @@
 // Copyright (c) Improbable Worlds Ltd, All Rights Reserved
 
-#pragma once
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/SpatialOSDispatcherInterface/SpatialOSDispatcherSpy.cpp:3]: ?

Comment (mironec) [reply SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/SpatialOSDispatcherInterface/SpatialOSDispatcherSpy.cpp:3]: Oh, I see, this is a cpp

File: SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp [hunk 2]
```diff
@@ -285,6 +285,17 @@ class SchemaValidator
...
 	int FreeId = 10000;
+
+	// This is needed to ensure the schema generated is the same for both Windows and macOS.
+	// The new-line characters differ which will fail the tests when running it on macOS.
+	FString CleanSchema(const FString& SchemaContent)
+	{
+		FString Content = SchemaContent;
+		Content.ReplaceInline(TEXT("\r"), TEXT(""));
+		Content.ReplaceInline(TEXT("\n"), TEXT(""));
+		Content.ReplaceInline(TEXT(" "), TEXT(""));
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp:296]: even spaces?

File: ci/setup-build-test-gdk.sh [hunk 3]
```diff
@@ -45,26 +44,25 @@ pushd "$(dirname "$0")"
...
-    # fi
+    if [[ -n "${SLOW_NETWORKING_TESTS}" ]]; then
+        echo "--- run-slow-networking-tests"
+        "${GDK_HOME}/ci/run-tests.sh" \
+            "${UNREAL_PATH}" \
+            "${TEST_PROJECT_NAME}" \
+            "${UPROJECT_PATH}" \
+            "SlowTestResults" \
+            "${TEST_REPO_MAP}" \
+            "+SpatialGDKSlow." \
```
Comment (mironec) [inline ci/setup-build-test-gdk.sh:65]:
```suggestion
            "SpatialGDKSlow." \
```
This should be enough, if launching two separate processes.
An alternative would be to call SpatialGDK.+SpatialGDKSlow. in one process if SLOW_NET_TEST

File: ci/setup-build-test-gdk.sh [hunk 4]
```diff
@@ -45,26 +44,25 @@ pushd "$(dirname "$0")"
...
-    #     "SpatialGDK+/Game/SpatialNetworkingMap" \
-    #     "True"
+    echo "--- run-fast-tests"
+    "${GDK_HOME}/ci/run-tests.sh" \
+        "${UNREAL_PATH}" \
+        "${TEST_PROJECT_NAME}" \
+        "${UPROJECT_PATH}" \
+        "TestResults" \
+        "${TEST_REPO_MAP}" \
+        ".SpatialGDK" \
```
Comment (mironec) [inline ci/setup-build-test-gdk.sh:54]:
```suggestion
        "SpatialGDK." \
```
I think

File: SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp (hunk 2 above)
Comment (jessicafalk) [reply SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp:296]: probably not, let me try to remove that line

**Pull request merged**